codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -468,6 +464,9 @@ public boolean write(int allowedBytes) {
                             size == bytesWritten && endOfStream, writePromise);
                 } while (size != bytesWritten && allowedBytes > bytesWritten);
                 size -= bytesWritten;
+                if (size == 0 && endOfStream) {
+                    lifecycleManager.closeLocalSide(stream, promise);
```

Comment:
As far as the contract goes for `error`, I think we only have to ensure that the flow controller calls the `error` method once.  If the encoder calls it on itself, that's his problem :)

WDYT?